shared/js/activeRoomStore.js

```diff
@@ -1178,7 +1183,7 @@ loop.store.ActiveRoomStore = (function(mozL10n) {
       // NOTE: when the window _is_ closed, hanging up the call is performed by
       //       MozLoopService, because we can't get a message across to LoopAPI
       //       in time whilst a window is closing.
-      if ((nextState === ROOM_STATES.FAILED || !this._isDesktop) && !failedJoinRequest) {
+      if ((nextState === ROOM_STATES.FAILED || windowStayingOpen || !this._isDesktop) && !failedJoinRequest) {
```

Comment:
nit: this is getting long. Probably split off `!failedJoinRequest` to a separate line.